### PR TITLE
add: OpenSearch snapshot restoration

### PR DESCRIPTION
**Pull request by harshini-rangaswamy**
## Describe your changes

Documentation for snapshot restoration in Aiven for OpenSearch. 
Ref: [MA-2784](https://aiven.atlassian.net/browse/MA-2784)

Preview: 
- https://harshini-os-snapshot-restora.aiven-docs.pages.dev/docs/products/opensearch/howto/migrate-external-snapshots-aiven-opensearch
- https://harshini-os-snapshot-restora.aiven-docs.pages.dev/docs/products/opensearch/howto/restore-snapshot-opensearch

## Checklist

- [x] The first paragraph of the page is on one line.
- [x] The other lines have a line break at 90 characters.
- [x] I checked the output.
- [x] I applied the [style guide](../styleguide.md).
- [x] My links start with `/docs/`.

[MA-2784]: https://aiven.atlassian.net/browse/MA-2784?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/products/opensearch/howto/restore-snapshot-opensearch.md [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+```bash
+curl -X PUT "https://SERVICE_NAME.aivencloud.com/_snapshot/my-snapshot-repo" \
+-H "Authorization: Bearer API_TOKEN" \
+-H "Content-Type: application/json" \
+-d '{
+  "type": "s3",
+  "settings": {
+    "bucket": "my-bucket",
+    "region": "us-west-2",
+    "access_key": "AWS_ACCESS_KEY",
```
Comment (willyborankin) [inline docs/products/opensearch/howto/restore-snapshot-opensearch.md:82]: `access_key` and `secret_key` are not part of the repo creation request

File: docs/products/opensearch/howto/restore-snapshot-opensearch.md [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+curl -X PUT "https://SERVICE_NAME.aivencloud.com/_snapshot/my-snapshot-repo" \
+-H "Authorization: Bearer API_TOKEN" \
+-H "Content-Type: application/json" \
+-d '{
+  "type": "azure",
+  "settings": {
+    "account": "AZURE_ACCOUNT_NAME",
+    "container": "my-container",
+    "base_path": "snapshots/",
+    "key": "AZURE_SECRET_KEY",
```
Comment (willyborankin) [inline docs/products/opensearch/howto/restore-snapshot-opensearch.md:205]: same account and key are not part of the PUT repository request

Comment (harshini-rangaswamy) [reply docs/products/opensearch/howto/restore-snapshot-opensearch.md]: removed this

File: docs/products/opensearch/howto/restore-snapshot-opensearch.md [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+
+- `PROJECT_NAME`: The name of your Aiven project.
+- `SERVICE_NAME`: The name of your Aiven OpenSearch service.
+- `repository`: The name of the snapshot repository.
+- `repository-type`: The repository type, set to `azure` for Microsoft Azure.
+- `container`: The Azure container name where the snapshot is stored.
+- `base_path`: The path within the container where the snapshot is stored.
+- `account`: The Azure account name.
+- `key`: The Azure secret key or SAS token.
+- `endpoint_suffix`: (Optional) Defines the DNS suffix for Azure Storage endpoints.
```
Comment (willyborankin) [inline docs/products/opensearch/howto/restore-snapshot-opensearch.md:227]: lets exclude `endpoint_suffix`

File: docs/products/opensearch/howto/restore-snapshot-opensearch.md [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+- `API_TOKEN`: The [API token for authentication](/docs/platform/howto/create_authentication_token).
+- `bucket`: The S3 bucket name where the snapshot is stored.
+- `region`: The AWS region where the bucket is located.
+- `endpoint`: (Optional) The service endpoint for S3-compatible services.
+
+</TabItem>
+<TabItem value="cli" label="CLI">
+
+```bash
+avn service elasticsearch-register-repository \
```
Comment (willyborankin) [inline docs/products/opensearch/howto/restore-snapshot-opensearch.md:94]: `avn service update -c s3_migration.region="asdsadasdasd" -c s3_migration.snapshot_name="dddd" -c s3_migration.base_path="/a/b/c/d" -c s3_migration.bucket="b" -c s3_migration.access_key="access_key" -c s3_migration.secret_key="secret_key_1" --project <p-name> <s_name>`

so `s3_migration.*` all properties mentioned above

Comment (willyborankin) [reply docs/products/opensearch/howto/restore-snapshot-opensearch.md:94]: And same for gcs and azure the difference is in the suffux:
`azure_migration.*` and `gcs_migration.*`

File: docs/products/opensearch/howto/restore-snapshot-opensearch.md [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+- **Google Cloud Storage (GCS)**:
+  - `credentials`: GCS credentials file content.
+  - `bucket`: GCS bucket name where the snapshot is stored.
+  - `base_path`: Path to repository data within the bucket.
+  - `snapshot_name`: Name of the snapshot to restore from.
+  - `compress`: (Optional) Whether to store metadata files in compressed format.
+  - `chunk_size`: (Optional) Size of chunks for large files during snapshot creation.
+
+- **Amazon S3 and S3-compatible services**:
+  - `region`: AWS/OCI region.
```
Comment (willyborankin) [inline docs/products/opensearch/howto/restore-snapshot-opensearch.md:39]: you forget settings for S3 credts

File: docs/products/opensearch/howto/restore-snapshot-opensearch.md [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+  - `bucket`: S3/OCI bucket name where the snapshot is stored.
+  - `base_path`: Path to repository data within the bucket.
+  - `snapshot_name`: Name of the snapshot to restore from.
+  - `endpoint`: (Optional) The service endpoint for S3-compatible services.
+  - `server_side_encryption`: (Optional) Whether to enable server-side encryption.
+  - `compress`: (Optional) Whether to store metadata files in compressed format.
+  - `chunk_size`: (Optional) Size of chunks for large files during snapshot creation.
+
+- **Microsoft Azure**:
+  - `container`: Azure container name where the snapshot is stored.
```
Comment (willyborankin) [inline docs/products/opensearch/howto/restore-snapshot-opensearch.md:49]: same there are no creds for Azure

File: docs/products/opensearch/howto/restore-snapshot-opensearch.md [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+```bash
+curl -X PUT "https://SERVICE_NAME.aivencloud.com/_snapshot/my-snapshot-repo" \
+-H "Authorization: Bearer API_TOKEN" \
+-H "Content-Type: application/json" \
+-d '{
+  "type": "gcs",
+  "settings": {
+    "bucket": "my-gcs-bucket",
+    "base_path": "snapshots/",
+    "credentials": "GCS_CREDENTIALS_FILE_CONTENT"
```
Comment (willyborankin) [inline docs/products/opensearch/howto/restore-snapshot-opensearch.md:135]: there is not such property for repo

File: docs/products/opensearch/howto/restore-snapshot-opensearch.md [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+```bash
+curl -X PUT "https://SERVICE_NAME.aivencloud.com/_snapshot/my-snapshot-repo" \
+-H "Authorization: Bearer API_TOKEN" \
+-H "Content-Type: application/json" \
+-d '{
+  "type": "azure",
+  "settings": {
+    "container": "my-container",
+    "base_path": "snapshots/",
+    "endpoint_suffix": "AZURE_ENDPOINT_SUFFIX"
```
Comment (willyborankin) [inline docs/products/opensearch/howto/restore-snapshot-opensearch.md:191]: could you plz remove it?

**Review by willyborankin — COMMENTED** [review]
All good but we need to discuss some settings

File: docs/products/opensearch/howto/restore-snapshot-opensearch.md (hunk 5 above)
Comment (harshini-rangaswamy) [reply docs/products/opensearch/howto/restore-snapshot-opensearch.md:39]: What is the setting for s3  credts?

File: docs/products/opensearch/howto/restore-snapshot-opensearch.md (hunk 4 above)
Comment (harshini-rangaswamy) [reply docs/products/opensearch/howto/restore-snapshot-opensearch.md:94]: Is this the correct command to register a repository?

File: docs/products/opensearch/howto/restore-snapshot-opensearch.md [hunk 9]
```diff
@@ -0,0 +1,386 @@
...
+-d '{
+  "type": "azure",
+  "settings": {
+    "container": "my-container",
+    "base_path": "snapshots/",
+    "account": "my-account",
+    "key": "your-key"
+  }
+}'
+```
```
Comment (axeoman) [inline docs/products/opensearch/howto/restore-snapshot-opensearch.md:191]: I don't think this steps (for each cloud provider) are needed and even possible from customer's machine. 
Everything including creation of repository is done by our automation, so customer should just update user-config parameters with credentials according to our current logic.

File: docs/products/opensearch/howto/restore-snapshot-opensearch.md [hunk 10]
```diff
@@ -0,0 +1,386 @@
...
+<TabItem value="cli" label="CLI">
+
+
+```bash
+avn service elasticsearch-check-restore-status \
+  --project PROJECT_NAME \
+  --service SERVICE_NAME \
+  --repository my-snapshot-repo \
+  --snapshot SNAPSHOT_NAME
+```
```
Comment (axeoman) [inline docs/products/opensearch/howto/restore-snapshot-opensearch.md:313]: This command does not exist in aiven-client, this is probably meant to be part of admin CLI which is not accessible by customer.

File: docs/products/opensearch/howto/restore-snapshot-opensearch.md [hunk 11]
```diff
@@ -0,0 +1,386 @@
...
+
+</TabItem>
+<TabItem value="cli" label="CLI">
+
+```bash
+avn service elasticsearch-list-indices \
+  --project PROJECT_NAME \
+  --service SERVICE_NAME
+```
```
Comment (axeoman) [inline docs/products/opensearch/howto/restore-snapshot-opensearch.md:337]: Same here - does not exist.

**Review by axeoman — CHANGES_REQUESTED** [review]

File: docs/products/opensearch/howto/migrate-opendistro-security-config-aiven.md [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+- You need the admin user key and certificate to migrate internal users and their
+  passwords. The password hash is fetched directly from the `.opendistro_security` index
+  because the standard API endpoint doesn’t return it.
+- The script assumes HTTPS is enabled on both the source and target services.
+
+:::
+
+## Migration steps
+
+To migrate your security configuration from Opendistro to Aiven for OpenSearch:
```
Comment (ArthurFlag) [inline docs/products/opensearch/howto/migrate-opendistro-security-config-aiven.md:52]: Semicolons should only introduce enumerations (lists). Section are meant to be potentially moved around/rearranged so cannot be introducted with semicolons.

File: docs/products/opensearch/howto/migrate-opendistro-security-config-aiven.md [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+    "port": target-port-number,
+    "password": "os-sec-admin-user-password"
+  }
+}
+```
+
+### Run the migration script
+
+Once your configuration file is ready, run the following command to execute the
+migration:
```
Comment (ArthurFlag) [inline docs/products/opensearch/howto/migrate-opendistro-security-config-aiven.md:82]: Let's never mention "the following commands", people know it is a command, and we have 2-4 words: 
```suggestion
Once your configuration file is ready, execute the migration:
```

File: docs/products/opensearch/howto/migrate-snapshot-data-opensearch.md [hunk 14]
```diff
@@ -0,0 +1,339 @@
...
+## Prerequisites
+
+Before you begin, ensure that:
+
+- You have an active Aiven account with a configured project.
+- A target Aiven for OpenSearch service within your project.
+- Your OpenSearch or Elasticsearch snapshot is from version 7.10.2 or earlier and is
+  compatible with your target service.
+- Enable the [feature preview](/docs/platform/howto/feature-preview) to try
+  migrating external snapshots.
```
Comment (ArthurFlag) [inline docs/products/opensearch/howto/migrate-snapshot-data-opensearch.md:33]: This bullet doesn't work with with your introduction sentence.

**Review by ArthurFlag — CHANGES_REQUESTED** [review]
Minor linting comments, rest looks gtm 👍

File: docs/products/opensearch/howto/migrate-snapshot-data-opensearch.md (hunk 14 above)
Comment (harshini-rangaswamy) [reply docs/products/opensearch/howto/migrate-snapshot-data-opensearch.md:33]: Thanks! This feature is behind ACL. Update the LA tag.

**Review by ArthurFlag — APPROVED** [review]

**Pull request merged**